### PR TITLE
Add nomad

### PR DESCRIPTION
**Pull request by jonashaag**
Things that could be improved: https://github.com/conda-forge/staged-recipes/pull/21541#issuecomment-1354409709

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jonashaag** [comment]
@fhoehle @xhochy any ideas about this?

```
2022-12-14T11:51:21.6521071Z # github.com/opencontainers/runc/libcontainer/nsenter
2022-12-14T11:51:21.6522300Z cloned_binary.c:67:6: warning: #warning "libc is outdated -- using hard-coded SYS_memfd_create" [-Wcpp]
2022-12-14T11:51:21.6522948Z    67 | #    warning "libc is outdated -- using hard-coded SYS_memfd_create"
2022-12-14T11:51:21.6523320Z       |      ^~~~~~~
2022-12-14T11:51:21.6523703Z cloned_binary.c: In function 'seal_execfd':
2022-12-14T11:51:21.6524236Z cloned_binary.c:385:46: error: 'O_PATH' undeclared (first use in this function)
2022-12-14T11:51:21.6524682Z   385 |                         newfd = open(fdpath, O_PATH | O_CLOEXEC);
2022-12-14T11:51:21.6525033Z       |                                              ^~~~~~
2022-12-14T11:51:21.6525462Z cloned_binary.c:385:46: note: each undeclared identifier is reported only once for each function it appears in
2022-12-14T11:51:21.6526002Z cloned_binary.c: In function 'try_bindfd':
2022-12-14T11:51:21.6526517Z cloned_binary.c:431:30: error: 'O_PATH' undeclared (first use in this function)
2022-12-14T11:51:21.6526944Z   431 |         ret = open(template, O_PATH | O_CLOEXEC);
2022-12-14T11:51:21.6527260Z       |                              ^~~~~~
```

**Comment by xhochy** [comment]
@jonashaag Try adding `sysroot_linux-64 2.17  # [linux64]` to `build`

**Comment by jonashaag** [comment]
Do you also happen to know about the osx-64 failure? xref https://github.com/conda-forge/go-feedstock/pull/147#issuecomment-1109961827?

**Comment by xhochy** [comment]
> Do you also happen to know about the osx-64 failure? xref [conda-forge/go-feedstock#147 (comment)](https://github.com/conda-forge/go-feedstock/pull/147#issuecomment-1109961827)?

Add this to `conda_build_config.yaml`:

```
MACOSX_SDK_VERSION:        # [osx and x86_64]
  - '10.14'                # [osx and x86_64]
MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
  - '10.14'                # [osx and x86_64]
  ```

**Comment by jonashaag** [comment]
```
40G	license-files
```

😮

**Comment by jonashaag** [comment]
Open problems:

- `make prerelease` doesn't work because of some Go error I don't understand (it updates protobuf specs)
- We only want `bin/nomad` to be installed but some part of the build procedure adds more binaries (and potentially other stuff?)
- Can't ship license files because for some reason `go-licenses` will generate 40 GB of licenses. Maybe some problem with filesystem recursion
- WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(140): Could not remove or rename - Gigantic amount of `/home/conda/staged-recipes/build_artifacts/nomad_1671179478120/work_moved_nomad-1.3.3-h0b41bf4_0_linux-64/gopath/pkg/mod/oss.indeed.com/go/libtime@v1.5.0/decay/backoff.go.  Please remove this file manually (you may need to reboot to free file handles)` warnings

**Comment by jonashaag** [comment]
This is ready for review, note some of the things that could be improved here: https://github.com/conda-forge/staged-recipes/pull/21541#issuecomment-1354409709

**Comment by jonashaag** [comment]
@conda-forge/go

**Comment by jonashaag** [comment]
@conda-forge/go

**Comment by jonashaag** [comment]
@conda-forge/go

**Comment by jonashaag** [comment]
@xhochy any tips how to make progress here?

**Comment by jonashaag** [comment]
@conda-forge/go @xhochy halp

**Comment by jonashaag** [comment]
Turns out I pinged the wrong group, sorry for the spam conda-forge/go members!

@conda-forge/staged-recipes

**Review by beckermr — CHANGES_REQUESTED** [review]
We cannot ignore the license failure here. IDK what to do precisely though.

**Comment by xhochy** [comment]
@jonashaag Let's paircode on this in future. Also, use a different medium to ping me. I don't monitor staged-recipes so much anymore and as a colleague you have different means of communication available 🤪

**Comment by xhochy** [comment]
I guess this needs https://github.com/conda-forge/go-feedstock/pull/176

File: recipes/nomad/build.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+export GO111MODULE=on
+make deps
+
+# XXX We should run "make prerelease" here instead but that fails for some reason
```
Comment (jonashaag) [inline recipes/nomad/build.sh:8]: @xhochy want to try `make prelease`?

**Review by jonashaag — COMMENTED** [review]
Great work :)

File: recipes/nomad/build.sh (hunk 1 above)
Comment (xhochy) [reply recipes/nomad/build.sh:8]: This isn't working as a lot of protobuf sources aren't part of the tarball. From the looks, the generated files were probably generated as part of the tarball creation process. Thus I think the commands here look correct.

**Review by jaimergp — APPROVED** [review]
LGTM.

I'll just bring to your attention (in case it was missed), that the line `make ember-dist static-assets` has a couple git errors on macos (but not on linux?), and the following steps seem to have a longer output via `npm`? Might be nothing but just wanted to make sure you knew about it.

The produced artifacts are reasonably sized (48MB total, with most of it being `bin/nomad`, as expected in a Go binary I guess).

**Pull request merged**